### PR TITLE
[PS2] Adding Joystick Driver and SDL_main

### PR DESCRIPTION
**Pull request by fjtrujy**
## Description
This PR continues with the PS2 port.
This time I'm adding:
- SDL2_main support
- Joystick driver

In order to be able to use easily I/O drivers in SDL, I have created another repository called `ps2_drivers` you can see that It was added to the CI. 
The idea is to be included in the official `ps2dev` docker, so once it is included we will remove it from the CI.

Cheers

### CODE REVIEW

File: test/basiccontrollermap.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*
```
Comment (madebr) [inline test/basiccontrollermap.c:1]: This file should be added to `test/basiscontrollermap.c` and probably also `test/Makefile.in`.

File: src/joystick/ps2/SDL_sysjoystick.c [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+struct JoyInfo {
+    uint8_t padBuf[256];
+    uint16_t btns;
+    uint8_t analog_state[PS2_TOTAL_AXIS];
+    uint8_t port;
+    uint8_t slot;
+    int8_t rumble_ready;
+} __attribute__ ((aligned (64)));
+
+static uint8_t __enabled_pads = 0;
```
Comment (madebr) [inline src/joystick/ps2/SDL_sysjoystick.c:57]: Variables starting with `_` are reserved for compiler and c library authors.

File: test/basiccontrollermap.c (hunk 1 above)
Comment (fjtrujy) [reply test/basiccontrollermap.c:1]: Oh sorry, I didn't want to add this test here. I was thinking to do it in a separate PR

File: src/joystick/ps2/SDL_sysjoystick.c (hunk 2 above)
Comment (fjtrujy) [reply src/joystick/ps2/SDL_sysjoystick.c]: Fixed!

File: test/basiccontrollermap.c (hunk 1 above)
Comment (fjtrujy) [reply test/basiccontrollermap.c]: File removed!

File: CMakeLists.txt [hunk 3]
```diff
@@ -2526,14 +2526,23 @@ elseif(PSP)
   endif(NOT SDL2_DISABLE_SDL2MAIN)
 
 elseif(PS2)
-  list(APPEND EXTRA_CFLAGS "-DPS2" "-D__PS2__")
+  list(APPEND EXTRA_CFLAGS "-DPS2" "-D__PS2__" "-I$ENV{PS2SDK}/ports/include")
```
Comment (madebr) [inline CMakeLists.txt:2529]: Just wondering, doesn't the ps2 cmake toolchain add `-DPS2` and `-D__PS2__` as compile flags?

Comment (fjtrujy) [reply CMakeLists.txt:2529]: No, they have currently just `_EE` but IMO is cleaner to have `PS2` and `_PS2_`.
I will try to convince PS2DEV guys to put there these flags, and then I will remove it from here

**Comment by fjtrujy** [comment]
@madebr I have created a separate PR #5840 for the test we were speaking before

**Comment by fjtrujy** [comment]
Let me know if something else needs to be applied here.

**Comment by fjtrujy** [comment]
Could this be merged? 
More drivers are coming 😃

**Comment by fjtrujy** [comment]
What about this PR @madebr ?
Cheers

**Comment by slouken** [comment]
FYI, the CI for this merge failed:
```
/usr/local/ps2dev/ee/lib/gcc/mips64r5900el-ps2-elf/11.3.0/../../../../mips64r5900el-ps2-elf/bin/ld: libSDL2.a(SDL_sysjoystick.c.obj): in function `PS2_JoystickInit':
SDL_sysjoystick.c:(.text+0xaec): undefined reference to `mtapPortOpen'
/usr/local/ps2dev/ee/lib/gcc/mips64r5900el-ps2-elf/11.3.0/../../../../mips64r5900el-ps2-elf/bin/ld: SDL_sysjoystick.c:(.text+0xafc): undefined reference to `mtapPortOpen'
/usr/local/ps2dev/ee/lib/gcc/mips64r5900el-ps2-elf/11.3.0/../../../../mips64r5900el-ps2-elf/bin/ld: /usr/local/ps2dev/ps2sdk/ports/lib/libps2_drivers.a(init_ps2_joystick_driver.o): in function `init_joystick_driver':
(.text+0x84): undefined reference to `mtapInit'
```

**Comment by madebr** [comment]
I re-ran the failing job and it succeeded on the 2nd try.
I suppose there must have been a temporary problem with the ps2dev docker image.
Looking at [its docker history](https://hub.docker.com/u/ps2dev), there was activity around the same time.

https://github.com/libsdl-org/SDL/runs/7079609067?check_suite_focus=true

**Pull request merged**